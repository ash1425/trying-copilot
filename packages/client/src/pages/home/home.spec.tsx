import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import userEvent from "@testing-library/user-event";
import * as reactRedux from "react-redux";
import * as middleware from "../../redux/Counter/counter.middleware";
import Home from "./home";

const mockStore = configureMockStore();

describe("Home page", () => {
  let store: any;
  let dummyDispatch: any;
  const useSelectorMock = jest.spyOn(reactRedux, "useSelector");
  const useDispatchMock = jest.spyOn(reactRedux, "useDispatch");
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
    dummyDispatch = jest.fn();
    useDispatchMock.mockReturnValue(dummyDispatch);
    jest.spyOn(middleware, "counterIncrement");
    jest.spyOn(middleware, "counterDecrement");
    jest.spyOn(middleware, "counterReset");

    store = mockStore({ counter: { count: 0 } });
  });
  test("when click on Hello World link, renders Hello World component", () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
  });

  test("when click on Hello World link, display text", () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    expect(screen.getByText("Welcome to my Home Page")).toHaveTextContent(
      "Welcome to my Home Page"
    );
  });
  test("Verifies the snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("when click on Increase Count button, dispatched the action and reflect the changes", () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    userEvent.click(screen.getByText("Increase Count"));
    expect(middleware.counterIncrement).toHaveBeenCalledTimes(1);
    expect(dummyDispatch).toHaveBeenCalled();
  });

  test("when click on Decrease Count button, dispatched the action and reflect the changes", () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    userEvent.click(screen.getByText("Decrease Count"));
    expect(middleware.counterDecrement).toHaveBeenCalledTimes(1);
    expect(dummyDispatch).toHaveBeenCalled();
  });

  test("when click on Reset button, dispatched the action and reflect the changes", () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>
    );

    userEvent.click(screen.getByText("Reset"));
    expect(middleware.counterReset).toHaveBeenCalledTimes(1);
    expect(dummyDispatch).toHaveBeenCalled();
  });
});
