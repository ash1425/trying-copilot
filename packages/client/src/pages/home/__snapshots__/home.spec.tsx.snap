// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Verifies the snapshot 1`] = `
<DocumentFragment>
  <main
    class="home-page"
  >
    <h1
      class="home-header"
    >
      Welcome to my Home Page
    </h1>
    <div
      class="counter"
    >
      <div
        class="counter-label"
      >
         Count: 0
      </div>
      <div
        class="counter-btn-container"
      >
        <div
          class="btn primary increment"
        >
          Increase Count
        </div>
        <div
          class="btn primary decrement"
        >
          Decrease Count
        </div>
        <div
          class="btn secondary reset"
        >
          Reset
        </div>
      </div>
    </div>
    <h2>
      Example application
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
    <div>
      <div
        class="accordion"
      >
        <div
          class="accordion-item"
        >
          <div
            class="accordion-title"
          >
            <div>
              section 1
            </div>
            <div>
              +
            </div>
          </div>
        </div>
      </div>
      <div
        class="accordion"
      >
        <div
          class="accordion-item"
        >
          <div
            class="accordion-title"
          >
            <div>
              section 2
            </div>
            <div>
              +
            </div>
          </div>
        </div>
      </div>
      <div
        class="accordion"
      >
        <div
          class="accordion-item"
        >
          <div
            class="accordion-title"
          >
            <div>
              section 3
            </div>
            <div>
              +
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          <div>
            <div
              class="carousel"
            >
              <div
                class="carousel-image"
              >
                <img
                  alt=""
                  id="image-1"
                  src="test-file-stub"
                  style="width: 100%; height: 400px;"
                />
              </div>
              <div
                class="carousel-controls"
              >
                <div
                  class="circle"
                >
                  <div
                    class="text"
                  >
                    prev
                  </div>
                </div>
                <div
                  class="circle"
                >
                  <div
                    class="text"
                  >
                    next
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div />
        <div />
        <div />
      </div>
    </div>
  </main>
</DocumentFragment>
`;
